kernel/src/cpuid.c

```diff
@@ -0,0 +1,38 @@
+#include <stdint.h>
+
+#include "io.h"
+#include "cpuid.h"
+
+void print_reg(uint32_t r) {
+    char register_val[5] = {};
+
+    // stores each byte from the register in reverse
+    for (int i = 3; i >= 0; i--) {
+        register_val[3 - i] = (r >> (CHAR_BIT * (3 - i))) & 0xFF;
+    }
+    register_val[4] = 0;
+
+    puts(register_val);
+}
+
+void print_vendor(uint32_t ebx, uint32_t ecx, uint32_t edx) {
```

Comment:
Missing const qualifier
```suggestion
void print_vendor(const uint32_t ebx, const uint32_t ecx, const uint32_t edx) {
```